src/deep-resource/lib/Resource/LambdaResponse.js

```diff
@@ -11,118 +11,117 @@ import {Response} from './Response';
-      let response = this._getPayload();
-
-      if (response && typeof response.errorMessage === 'undefined') {
-        this._data = response;
-      }
+      this._data = responsePayload;
     }
-
-    return this._data;
   }
 
   /**
-   * @returns {String}
+   * @param {Object|null} responsePayload
+   * @private
    */
-  get error() {
-    if (this._error) {
-      return this._error;
-    }
-
+  _fillError(responsePayload) {
     if (this._rawError) {
       this._error = this._rawError;
-    } else {
-      if (!this._request.async) {
-        let response = this._getPayload();
-
-        if (response && typeof response.errorMessage !== 'undefined') {
-          this._error = response.errorMessage;
-        }
-      } else {
-        this._error = 'Unknown async invocation error';
+    } else if (!this._request.async) {
+      if (!responsePayload) {
+        this._error = new Error('There is no error nor payload in Lambda response');
+      } else if (responsePayload.hasOwnProperty('errorMessage')) {
+        this._error = LambdaResponse.getPayloadError(responsePayload);
       }
+    } else if (this._statusCode !== 202) { // check for failed async invocation
+      this._error = new Error('Unknown async invocation error');
     }
-
-    return this._error;
   }
 
   /**
-   * @returns {String}
+   * @private
    */
-  get errorType() {
-    if (this._errorType) {
-      return this._errorType;
-    }
-
-    if (this._rawError) {
-      this._errorType = (this._rawError && this._rawError.name) ? this._rawError.name : 'Error';
-    } else {
-      if (!this._request.async) {
-        let response = this._getPayload();
-
-        if (response && typeof response.errorType !== 'undefined') {
-          this._errorType = response.errorType;
-        } else {
-          this._errorType = 'Error';
-        }
-      } else {
-        this._errorType = 'AsyncInvocationError';
-      }
-    }
-
-    return this._errorType;
+  _fillStatusCode() {
+    this._statusCode = parseInt(this._rawData.StatusCode || this._rawData.Status || 500);
   }
 
   /**
-   * @returns {String}
+   * @returns {Object|null}
+   * @private
    */
-  get statusCode() {
-    if (this._statusCode) {
-      return this._statusCode;
-    }
+  _decodePayload() {
+    if (typeof this._rawData === 'object' &&
+      this._rawData.hasOwnProperty('Payload')) {
+      let payload = this._rawData.Payload;
 
-    if (this._rawData) {
-      this._statusCode = this._rawData.StatusCode || this._rawData.Status;
+      return typeof payload === 'string' ? JSON.parse(payload) : payload;
     }
 
-    return this._statusCode;
+    return null;
   }
 
   /**
-   * @returns {Object|null}
-   * @private
+   * @param {Object} payload
+   * @returns {Error|null}
    */
-  _getPayload() {
-    if (typeof this._rawData === 'object' &&
-      this._rawData.hasOwnProperty('Payload')) {
-      let payload = this._rawData.Payload;
+  static getPayloadError(payload) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (9)](https://www.codacy.com/app/MitocGroup/deep-framework/file/1564884016/issues/source?bid=2895475&fileBranchId=3020199#l85)